### PR TITLE
View video page mostly functional

### PR DESCRIPTION
**Pull request by arthurmcgregor**
Work in progress towards #2. Now has most functionality and is usable I believe. I would like to write unit tests before moving on to something else while it is still fresh in my mind, but don't know how to approach that. Suggestions @adbs1 @mjs ??

PS to get to this page, hit the 'View' button for any recording on the recordings page.

Not yet implemented:
- Testing & bug checking
- Keyboard shortcuts
- 'Me!' displayed on own tags
- Additional info display
- Confirm button to AI tags
- Saving default tags
- Handle no recording
(may be more?!)

### CODE REVIEW

File: src/api/Recording.api.js [hunk 1]
```diff
@@ -33,17 +52,31 @@ function id(id, token) {
...
+      method:"PATCH",
+      headers: {'Authorization': token, 'Content-Type': 'application/x-www-form-urlencoded'},
+      mode: 'cors',
+      cache: "no-cache",
+      body: body
+    }
+  );
+}
+
+function del(id, token) {
```
[the remark this answers is not in the document]
Comment (arthurmcgregor) [reply src/api/Recording.api.js]: delete was a reserved word
https://www.w3schools.com/js/js_reserved.asp

File: src/api/Tag.api.js [hunk 2]
```diff
@@ -0,0 +1,45 @@
+import fetch from 'cross-fetch';
+import { Config } from '../../app.config' // eslint-disable-line
+
+export default {
+  addTag, deleteTag
+};
+
+let tagApi = '/api/v1/tags';
```
[the remark this answers is not in the document]
Comment (arthurmcgregor) [reply src/api/Tag.api.js]: done

File: src/components/Video/ObservedAnimals.vue [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+      ]
+    };
+  },
+  // https://vuejs.org/v2/style-guide/#Simple-computed-properties-strongly-recommended
+  computed: {
+  },
+  methods: {
+    animalImage: function (animal, event) {
+      // Struggling to get images to show correctly so using work-around
+      // suggeested at bottom of this page
```
[the remark this answers is not in the document]
Comment (arthurmcgregor) [reply src/components/Video/ObservedAnimals.vue]: whoops. done

File: src/components/Video/ObservedAnimals.vue [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+        {key: 'deleteButton', label: ''}
+      ]
+    };
+  },
+  // https://vuejs.org/v2/style-guide/#Simple-computed-properties-strongly-recommended
+  computed: {
+  },
+  methods: {
+    animalImage: function (animal, event) {
+      // Struggling to get images to show correctly so using work-around
```
[the remark this answers is not in the document]
Comment (arthurmcgregor) [reply src/components/Video/ObservedAnimals.vue]: done

File: src/components/Video/ObservedAnimals.vue [hunk 5]
```diff
@@ -0,0 +1,149 @@
...
+        return `<img class="animal-image" src="${humanLink}" />`;
+      case "bird":
+        return `<img class="animal-image" src="${birdLink}" />`;
+      case "bird/kiwi":
+        return `<img class="animal-image" src="${kiwiLink}" />`;
+      case "unidentified":
+        return `<img class="animal-image" src="${noneLink}" />`;
+      default:
+        return null;
+      }
```
[the remark this answers is not in the document]
Comment (arthurmcgregor) [reply src/components/Video/ObservedAnimals.vue]: Much nicer! Thanks. Done

File: src/components/Video/PrevNext.vue [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+  },
+  // https://vuejs.org/v2/style-guide/#Component-data-essential
+  data () {
+    return {
+    };
+  },
+  // https://vuejs.org/v2/style-guide/#Simple-computed-properties-strongly-recommended
+  computed: {
+
+  },
```
[the remark this answers is not in the document]
Comment (arthurmcgregor) [reply src/components/Video/PrevNext.vue]: Good point. I've removed unneeded boilerplate from all of the views/components that I've written for cleanliness

File: src/api/Recording.api.js [hunk 7]
```diff
@@ -20,6 +20,25 @@ function query(token, limit, offset, tagMode, tags, query) {
...
+    tagMode: tagMode
+  };
+  if (order) {
+    params.order = order;
+  }
+  let queryString = Object.keys(params).map((key) => {
+    return encodeURIComponent(key) + '=' + encodeURIComponent(params[key]);
+  }).join('&');
+  let url = `${Config.api}` + recordingApi + "?" + queryString;
+  return url;
```
[the remark this answers is not in the document]
Comment (arthurmcgregor) [reply src/api/Recording.api.js]: Yup that'd make life easier.... something like this? https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams/append Still fairly new though?

**Comment by mjs** [comment]
This is looking great to me but I still think it would be good for @adbs1 to take a look too, if he's able.

File: src/api/Recording.api.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (mjs) [reply src/api/Recording.api.js]: Ah of course

**Comment by adbs1** [comment]
Looks good to me also. Early commits so plenty to refactor as we move forward and learn best practice  in Vue / Vuex. On the testing, Karma + Jasmine is a good option which I'll set up and commit asap. Starting with the low hangers first, api and stores, the vues templates will be more involved.

**Pull request merged**